src/components/Contexts/StepperDialogProvider.jsx

```diff
@@ -1,27 +1,39 @@
-import React, { createContext, useCallback, useState } from 'react'
+import React, { createContext, useCallback, useEffect, useState } from 'react'
 
 const StepperDialogContext = createContext()
 
 const StepperDialogProvider = ({ children }) => {
   const [isStepperDialogOpen, setIsStepperDialogOpen] = useState(false)
   const [stepperDialogTitle, setStepperDialogTitle] = useState('')
   const [allCurrentPages, setAllCurrentPages] = useState([])
-  const [currentPage, setCurrentPage] = useState(1)
+  const [allCurrentPagesSafed, setAllCurrentPagesSafed] = useState([])
+  const [currentPageIndex, setCurrentPageIndex] = useState(1)
+
+  useEffect(() => {
+    if (allCurrentPages.length > 0) {
+      const temp = allCurrentPages
+        .flatMap(page => new Array(page.occurrence || 1).fill(page))
```

Comment:
Shouldn't pages be sorted by pageIndex first? `pageIndex` from the json file seems not to be used.